mobile-app/app/screens/AppNavigator/screens/Balances/BalancesScreen.tsx

```diff
@@ -241,8 +252,57 @@ export function BalancesScreen ({ navigation }: Props): JSX.Element {
     }
   ]
 
+  // Asset sort bottom sheet list
+  const [assetSortType, setAssetSortType] = useState<BalancesSortType>(BalancesSortType.HighestDenominationValue) // to display selected sorted type text
+  const [isSorted, setIsSorted] = useState<boolean>(false) // to display acsending/descending icon
+  const [hideIcon, setHideIcon] = useState(false)
+  const [showAssetSortBottomSheet, setShowAssetSortBottomSheet] = useState(false)
+  const modifiedDenominationCurrency = useMemo(() => denominationCurrency === 'USDT' ? 'USD' : denominationCurrency, [denominationCurrency])
+  const sortTokensAssetOnType = useCallback((assetSortType: BalancesSortType): BalanceRowToken[] => {
+    let sortTokensFunc: (a: BalanceRowToken, b: BalanceRowToken) => number
+    switch (assetSortType) {
+      case (BalancesSortType.HighestDenominationValue):
+        sortTokensFunc = (a, b) => b.usdAmount.minus(a.usdAmount).toNumber()
+        break
+      case (BalancesSortType.LowestDenominationValue):
+        sortTokensFunc = (a, b) => a.usdAmount.minus(b.usdAmount).toNumber()
+        break
+      case (BalancesSortType.HighestTokenAmount):
+        sortTokensFunc = (a, b) => new BigNumber(b.amount).minus(new BigNumber(a.amount)).toNumber()
+        break
+      case (BalancesSortType.LowestTokenAmount):
+        sortTokensFunc = (a, b) => new BigNumber(a.amount).minus(new BigNumber(b.amount)).toNumber()
+        break
+      case (BalancesSortType.AtoZ):
+        sortTokensFunc = (a, b) => a.symbol.localeCompare(b.symbol)
+        break
+      case (BalancesSortType.ZtoA):
+        sortTokensFunc = (a, b) => b.symbol.localeCompare(a.symbol)
+        break
+      default:
+        sortTokensFunc = (a, b) => b.usdAmount.minus(a.usdAmount).toNumber()
+    }
+
+    return filteredTokens.sort(sortTokensFunc)
+  }, [filteredTokens, assetSortType, denominationCurrency])
+
+  useEffect(() => {
+    setAssetSortType(BalancesSortType.HighestDenominationValue) // reset sorting state upon denominationCurrency change
+  }, [denominationCurrency])
+
+  // conditions to display sort icons
+  useEffect(() => {
+    if (assetSortType.includes('Lowest')) {
+      setIsSorted(true)
+    } else if (assetSortType.includes('A to Z') || assetSortType.includes('Z to A')) {
```

Comment:
can replace this if else loop with return also, with if else loop code readability decreases

Reply:
@fullstackninja864 Let's resolve this on another PR @chloezxyy @pierregee